### PR TITLE
[NSA-7492] fix redirect urls limitation

### PR DESCRIPTION
**Pull request by ipredoi**
Jira ticket: 
https://dfe-secureaccess.atlassian.net/browse/NSA-7492

Changes: 

- added functionality to save Redirect URLs and PostReset URLs for service changes in the application local storage, enabling their accessibility on the 'Review service configuration changes' page. This enables the addition of an unlimited number of URLs in one go without crashing the session.
- added functionality to clear the links when form is submitted / or after 1h 
- updated the unit tests to cover the new functionality

### CODE REVIEW

File: src/infrastructure/utils/serviceConfigCache.js [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+
+    results.forEach((message) => {
+      logger.info(message);
+    });
+
+    return true;
+  } catch (error) {
+    logger.error(`Error while removing ${keyWord} data from local storage:`, error);
+    throw error;
+  }
```
Comment (malikimran34) [inline src/infrastructure/utils/serviceConfigCache.js:53]: Add a logger.info that delete was success

File: src/infrastructure/utils/serviceConfigCache.js [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+  } catch (error) {
+    logger.error(`Error while passing ${redirectUrlsKey} data to local storage for service ID ${serviceId}:`, error);
+  }
+};
+
+const retreiveRedirectUrlsFromStorage = async (redirectUrlsKey, serviceId) => {
+  try {
+    const result = await localStorage.getItem(redirectUrlsKey);
+    if (result) {
+      logger.info(`${redirectUrlsKey} data retrieved successfully from local storage for service ID: ${serviceId}`);
```
Comment (malikimran34) [inline src/infrastructure/utils/serviceConfigCache.js:19]: shorten the message please e.g. {service name}: url

File: src/app/services/serviceConfig.js [hunk 3]
```diff
@@ -70,8 +97,10 @@ const buildCurrentServiceModel = async (req) => {
 
 const getServiceConfig = async (req, res) => {
   try {
-    if (req.session.serviceConfigurationChanges && req.query.action !== ACTIONS.AMEND_CHANGES) {
+    if (req.query.action !== ACTIONS.AMEND_CHANGES) {
       req.session.serviceConfigurationChanges = {};
```
Comment (malikimran34) [inline src/app/services/serviceConfig.js:101]: log.info amend changes action please

**Review by malikimran34 — APPROVED** [review]

File: src/app/services/serviceConfig.js (hunk 3 above)
Comment (ipredoi) [reply src/app/services/serviceConfig.js:101]: the query parameter is used to populate the fields with the values from Review service configuration page when the user is coming back from Review service configuration page - there will be no need to log.info this action

**Pull request merged**